src/ert/scheduler/scheduler.py

```diff
@@ -190,13 +190,20 @@ async def _publisher(self) -> None:
             ping_interval=60,
             close_timeout=60,
         ):
-            while True:
-                event = await self._events.get()
-                if event == CLOSE_PUBLISHER_SENTINEL:
-                    self._publisher_done.set()
-                    return
-                await conn.send(event)
-                self._events.task_done()
+            try:
+                while True:
+                    event = await self._events.get()
+                    if event == CLOSE_PUBLISHER_SENTINEL:
+                        self._publisher_done.set()
+                        return
+                    await conn.send(event)
+                    self._events.task_done()
+                    event = None
+            except ConnectionClosed:
+                logger.debug("Connection to EnsembleEvalutor went down, reconnecting.")
+                if event is not None:
+                    await self._events.put(event)
```

Comment:
Since asyncio.Queue is a FIFO queue, the order can get mixed when this happens, but this the EnsembleEvaluator should be able to deal with.

Reply:
Not sure with this one as (as we had seen before) the realizations could have been finished already and thus RUNNING / PENDING event can come after all's been done.

Reply:
Fixed now so that the order is kept.